src/Spago/Command/Resolution.purs

```diff
@@ -0,0 +1,45 @@
+module Spago.Command.Resolution where
+
+import Prelude
+
+import Data.Map (Map, lookup, toUnfoldable)
+import Data.Maybe (Maybe(..))
+import Data.String (joinWith)
+import Data.Tuple (Tuple(..))
+import Registry.PackageName (PackageName)
+import Registry.PackageName as PackageName
+import Registry.Range (Range)
+import Registry.Range as Range
+import Registry.Version (Version)
+import Registry.Version as Version
+import Spago.Config (Package(..))
+
+type VersionResolution =
+  { name :: PackageName
+  , requested :: Range
+  , resolved :: Version
+  }
+
+resolutions :: Map PackageName Package -> Map PackageName Range -> Array VersionResolution
+resolutions registry dependencies = do
+  Tuple name requested <- toUnfoldable dependencies
```

Comment:
I always find the monad instance of `Array` confusing - could we rewrite this function e.g. with `map`?